lib/web/join_tokens.go

```diff
@@ -57,26 +59,39 @@ type scriptSettings struct {
 	appURI         string
 }
 
-// createTokenRequest is the expected request body of
-// the endpoint to create token
-type createTokenRequest struct {
-	Roles types.SystemRoles `json:"roles"`
-}
-
 func (h *Handler) createTokenHandle(w http.ResponseWriter, r *http.Request, params httprouter.Params, ctx *SessionContext) (interface{}, error) {
-	var req createTokenRequest
+	var req types.ProvisionTokenSpecV2
```

Comment:
do we need to consider backward compatibility for this endpoint? wondering if the webui is the only user or we could have some customers calling it

Reply:
This endpoint is new, it was added in https://github.com/gravitational/teleport/pull/10914, so I assume no one uses it.

There is an old endpoint that the web UI used to call, it is been deprecated and will be deleted in v11, just in case some customer uses it.

Reply:
I didn't realize it was so new, not a concern then

Reply:
Btw I was taking a look at this again and noticed this change **is** backward compatible, since both `createTokenRequest` and `types.ProvisionTokenSpecV2` have a `Roles` property.

This PR only adds the capability of requesting different join methods.

To be sure I tested locally with teleport running this branch and webapps running master and it worked fine.

Sorry about the confusion.